### PR TITLE
optimize memory usage to remove test failures

### PR DESCRIPTION
**Pull request by jtnord**
includes #479 as needed to try and get a full CI run.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -75,7 +75,7 @@
       <dependency>
         <groupId>io.jenkins.tools.bom</groupId>
         <artifactId>bom-2.319.x</artifactId>
-        <version>1013.vf8058992a042</version>
+        <version>1090.v0a_33df40457a_</version>
```
Comment (jtnord) [inline pom.xml:78]: required to have a compatible set of plugins with the upgraded `git-plugin`

File: pom.xml [hunk 2]
```diff
@@ -87,7 +87,7 @@
       <dependency>
         <groupId>org.jenkins-ci.main</groupId>
         <artifactId>jenkins-test-harness</artifactId>
-	<version>1692.v41c59b_73de66-SNAPSHOT</version>
+	<version>1692.v41c59b_73de66</version>
```
Comment (jglick) [inline pom.xml:90]:
```suggestion
        <version>1692.v41c59b_73de66</version> <!-- TODO pending something currently unreleased -->
```

**Comment by jtnord** [comment]
build succeeded on Linux (11,8) windows build timedout during testing.
however it took over 40 minutes for maven (in the linux build) to get the reactor order and so I put the timeout here down to artifactory being awfully slow again.  just need a healthy speedy repository to show there are actually no regressions...

looking at the output of the Linix build for the plugin module

```
[2022-01-21T10:20:25.122Z] Running test
... snip
[2022-01-21T10:45:15.095Z] [INFO] Results:
[2022-01-21T10:45:15.095Z] [INFO] 
[2022-01-21T10:45:15.095Z] [WARNING] Tests run: 1406, Failures: 0, Errors: 0, Skipped: 21
```

it took 25 minutes to run the tests which includes ones that had previous been ignored, so I would like to call this good.

File: pipeline-model-definition/pom.xml [hunk 3]
```diff
@@ -255,6 +256,7 @@
       <artifactId>git</artifactId>
       <classifier>tests</classifier>
       <scope>test</scope>
+      <version>4.10.3</version>
```
[the remark this answers is not in the document]
Comment (jtnord) [reply pipeline-model-definition/pom.xml]: bom has been updated

File: pom.xml [hunk 4]
```diff
@@ -84,6 +84,11 @@
         <artifactId>jenkins-test-harness-tools</artifactId>
         <version>2.2</version>
       </dependency>
+      <dependency>
+        <groupId>org.jenkins-ci.main</groupId>
+        <artifactId>jenkins-test-harness</artifactId>
+        <version>1692.v41c59b_73de66</version>
```
Comment (jtnord) [inline pom.xml:90]: jenkinsci/jenkins-test-harness#380

File: pom.xml [hunk 5]
```diff
@@ -111,11 +116,6 @@
         <artifactId>pipeline-milestone-step</artifactId>
         <version>1.3.2</version>
       </dependency>
-      <dependency>
-        <groupId>org.jenkins-ci.plugins</groupId>
-        <artifactId>ant</artifactId>
-        <version>1.13</version>
-      </dependency>
```
Comment (jtnord) [inline pom.xml:118]: now in [the bom](https://github.com/jenkinsci/bom/blob/231c8d6403cf235a1f149fd24db857395d510cad/bom-weekly/pom.xml#L151-L155)

File: pom.xml [hunk 6]
```diff
@@ -75,7 +75,7 @@
       <dependency>
         <groupId>io.jenkins.tools.bom</groupId>
         <artifactId>bom-2.319.x</artifactId>
-        <version>1090.v0a_33df40457a_</version>
+        <version>1117.v62a_f6a_01de98</version>
```
Comment (jtnord) [inline pom.xml:78]: picks up an updated git plugin (4.10.3) that does not use local git templates when init'ing a repo
jenkinsci/git-plugin#1207

File: pom.xml (hunk 4 above)
Comment (jtnord) [reply pom.xml:90]:
```suggestion
        <version>1697.v6b_1e34cb_4ee6</version>
```

File: pom.xml [hunk 7]
```diff
@@ -84,6 +84,11 @@
         <artifactId>jenkins-test-harness-tools</artifactId>
         <version>2.2</version>
       </dependency>
+      <dependency>
+        <groupId>org.jenkins-ci.main</groupId>
+        <artifactId>jenkins-test-harness</artifactId>
+        <version>1697.v6b_1e34cb_4ee6</version>
+      </dependency>
```
Comment (raul-arabaolaza) [inline pom.xml:91]: This is defined in the `dependencyManagement` of the parent pom, so you can remove the version and instead override https://github.com/jenkinsci/plugin-pom/blob/bd11e866db97ccf017b504d64f4821d897ea8987/pom.xml#L67 for coherence of usage with other plugins

Comment (jtnord) [reply pom.xml:91]: given it was released today I highly doubt it :)

**Review by raul-arabaolaza — APPROVED** [review]

File: pom.xml [hunk 8]
```diff
@@ -84,6 +84,11 @@
         <artifactId>jenkins-test-harness-tools</artifactId>
         <version>2.2</version>
       </dependency>
+      <dependency>
```
Comment (jtnord) [inline pom.xml:87]:
```suggestion
      <!-- TODO remove this dependency when it is available in the plugin-pom -->
      <dependency>
```

**Review by jglick — APPROVED** [review]

**Review by car-roll — APPROVED** [review]

**Pull request merged**